lib/core/versions/latest/RequestHandler.ts

```diff
@@ -34,19 +35,26 @@ export default class RequestHandler implements IRequestHandler {
     console.info(`Handling operation request of size ${request.length} bytes...`);
 
     // Perform common validation for any write request and parse it into an `OperationModel`.
-    let operation: OperationModel;
+    let operationModel: OperationModel;
     try {
-      // Validate operation request size.
-      if (request.length > ProtocolParameters.maxOperationByteSize) {
-        const errorMessage = `Operation byte size of ${request.length} exceeded limit of ${ProtocolParameters.maxOperationByteSize}`;
-        console.info(errorMessage);
-        throw new SidetreeError(ErrorCode.OperationExceedsMaximumSize, errorMessage);
+      const operationRequest = await JsonAsync.parse(request);
+
+      // Check `operationData` property data size if they exist in the operation.
+      if (operationRequest.type === OperationType.Create ||
+          operationRequest.type === OperationType.Recover ||
+          operationRequest.type === OperationType.Update) {
+        const operationDataBuffer = Buffer.from(operationRequest.operationData);
+        if (operationDataBuffer.length > ProtocolParameters.maxOperationDataByteSize) {
+          const errorMessage = `operationDdata byte size of ${operationDataBuffer.length} exceeded limit of ${ProtocolParameters.maxOperationDataByteSize}`;
+          console.info(errorMessage);
+          throw new SidetreeError(ErrorCode.RequestHandlerOperationDataExceedsMaximumSize, errorMessage);
+        }
       }
 
-      operation = await Operation.parse(request);
+      operationModel = await Operation.parse(request);
 
       // Reject operation if there is already an operation for the same DID waiting to be batched and anchored.
-      if (await this.operationQueue.contains(operation.didUniqueSuffix)) {
+      if (await this.operationQueue.contains(operationModel.didUniqueSuffix)) {
         throw new SidetreeError(ErrorCode.QueueingMultipleOperationsPerDidNotAllowed);
```

Comment:
Small suggestion: if it is not too much work (especially for the tests), can we add a message to this error with actual did causing this error?

Reply:
Done.